### ISSUE TITLE
Does hyperparam of MLflow support TensorFlow?

### ISSUE THREAD

**Issue by lyw615** [mlflow/mlflow#1586]
## Describe the problem

I want to try the function of hyperparams tune in Tensorflow,but example of hyperparam just give example of Keras. If it supports Tensorflow,the usage :
self._best_model.set_weights([x.copy() for x in self.model.get_weights()]) 
in line 100 examples/hyperparam/train.py  should be replaced by which. Or the usage is same to Keras?

**Comment by ankitmathur-db** [comment]
That tutorial is written using `mlflow.keras` for Keras and tf.Keras workflows. We also have `mlflow.tensorflow`, which means you'd have to update line 77 to use `mlflow.tensorflow.log_model`.

However, the code from lines 100-102 are up to you to write - they work easily for most keras models using that code, but Tensorflow code is likely to depend on the specifics of the model you want to run (see https://github.com/tensorflow/tensorflow/issues/14726). `self.best_model` is just storing a clone of the model - Tensorflow has other mechanisms to do this, often specific to the specific model you're trying to save. Lines 100-102 reflect how you do this in Keras. Moreover, this is an optimization - we do this so that you don't constantly have to be writing relatively large models to disk - you could theoretically (at the risk of performance loss), call one of `mlflow.tensorflow.log_model`/`mlflow.tensorflow.save_model` at the end of every N epochs.
